server/src/api/v1/users/auth-user.ts

```diff
@@ -1,28 +1,45 @@
 import {Request, Response} from 'express';
 import {wrapResponse} from '../../../functions/response-wrapper';
-import {UserData} from '../../../interfaces/users.interface';
-import {mapUser} from '../../../functions/map-users.func';
+import {UserLoginData} from '../../../interfaces/users.interface';
 import {User} from '../../../models/user.model';
 import * as bcrypt from 'bcryptjs';
 import {jwtSign} from '../../../functions/jwt-sign.func';
 
 export async function loginUser(req: Request, res: Response): Promise<Response> {
 
-    const incomingData: UserData = req.body;
-    const mappedIncomingData: UserData = await mapUser(incomingData);
+    const incomingData: UserLoginData = req.body;
 
-    let success = true;
+    if (incomingData.password === undefined || (incomingData.email === undefined && incomingData.username === undefined)){
+        return res.status(400).send(wrapResponse(false, { error: 'Not all required fields have been set' }));
+    }
 
-    const user = await User.findOne(
-        {
-            where: {
-                email: mappedIncomingData.email,
-            }
-        })
-        .catch(() => {
-            success = false;
-            return null;
-        });
+    let success = true;
+    let user: User | null;
+    if (incomingData.email !== undefined) {
+        user = await User.findOne(
+            {
+                where: {
+                    email: incomingData.email,
+                }
+            })
+            .catch(() => {
+                success = false;
+                return null;
+            });
+    } else if (incomingData.username !== undefined) {
+        user = await User.findOne(
+            {
+                where: {
+                    username: incomingData.username,
+                }
+            })
+            .catch(() => {
+                success = false;
+                return null;
+            });
```

Comment:
Zu viel duplicate code - am Besten aufteilen in 'Query zusammenbauen' und dann 'Query ausführen' (ungetestet):

```suggestion
    const selectStatement = {
        where: {
        ...(incomingData?.email ? {email: incomingData.email} : {}),
        ...(incomingData?.username ? {email: incomingData.username} : {}),
        }
    };
    
        user = await User.findOne(selectStatement)
            .catch(() => {
                success = false;
                return null;
            });
```

Bitte nicht direkt commiten - in einem Kommentar programmieren ist doof :)